### PR TITLE
Cody: Adding github gitlab and google SSO

### LINKED ISSUE #19: JetBrains: move to social SSO onboarding
In VS Code, we link directly to GitHub/Google during onboarding instead of having a button that says "Sign in for free on sourcegraph.com". The results of this experiments have been positive enough that we should make the same change in the JetBrains plugin.

### PR DESCRIPTION
**Pull request by Sa1to**
It fixes #19 
## Test plan
* Sign out of all cody accounts
* Open Cody chat window
* Try to sign in with GitHub/GitLab/Google account

https://github.com/sourcegraph/jetbrains/assets/9321940/ea28bfd8-144f-4f5a-bcba-aa61512be09f

### CODE REVIEW

**Review by olafurpg — COMMENTED** [review]
I just tried this locally and confirmed that it works for Google, GitHub, and GitLab. This is awesome 🤩 

I haven't reviewed the code so it's good to get a second eye on the diff.

**Comment by Sa1to** [comment]
I just noticed one problem when trying to login via Sourcegraph from settings, fixing it atm.

**Comment by Sa1to** [comment]
Done :)

File: src/main/resources/icons/signIn/sign-in-logo-github.svg
Comment (mkondratek) [inline src/main/resources/icons/signIn/sign-in-logo-github.svg:1]: Does it look good with the light theme?

(the same question for the other icons)

File: src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt [hunk 1]
```diff
@@ -25,42 +25,44 @@ import com.sourcegraph.cody.ui.UnderlinedActionLink
...
+import javax.swing.JButton
 import javax.swing.JLabel
 import javax.swing.JPanel
 import javax.swing.border.Border
 
 class SignInWithSourcegraphPanel(private val project: Project) : JPanel() {
 
-  private val mainButton = UIComponents.createMainButton("Sign in for free with Sourcegraph.com")
+  private val signInWithGithubButton =
+      UIComponents.createMainButton("Sign in with GitHub", Icons.SignIn.Github)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt]: I see that for now "Sign in with GitHub" is used twice in this diff. 
Would it be possible to bring the texts into `SsoAuthMethod` enum?

(the same concern about all `"Sign in with .*"` ofc 😃)

File: src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt [hunk 2]
```diff
@@ -79,18 +81,47 @@ class SignInWithSourcegraphPanel(private val project: Project) : JPanel() {
...
+              ActionManager.getInstance(),
+              it.modifiers)
+      if (ActionUtil.lastUpdateAndCheckDumb(logInToSourcegraphAction, event, false)) {
+        ActionUtil.performActionDumbAwareWithCallbacks(logInToSourcegraphAction, event)
+      }
+    }
+    return functionGithub
+  }
+
   fun addMainButtonActionListener(actionListener: ActionListener) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt]: so now sign  in with github button is the main button - my question is:
what do we mean by the main button?

File: src/main/kotlin/com/sourcegraph/cody/chat/UIComponents.kt [hunk 3]
```diff
@@ -17,4 +18,11 @@ object UIComponents {
     button.setUI(buttonUI)
     return button
   }
+
+  @JvmStatic
+  fun createMainButton(text: String, icon: Icon): JButton {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/UIComponents.kt:23]: if the only usage of `createMainButton` is this case, and if we were able to bring both the text and the icon into the enum, this method could take only the one enum argument - I do not see the wider scope here, but pls take it into consideration

File: src/main/kotlin/com/sourcegraph/cody/config/BaseLoginDialog.kt [hunk 4]
```diff
@@ -62,8 +64,9 @@ abstract class BaseLoginDialog(
     Disposer.register(disposable) { emptyProgressIndicator.cancel() }
 
     startGettingToken()
+    //      HERE FOR SURE
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/BaseLoginDialog.kt]: to be removed I guess 😅

**Review by mkondratek — CHANGES_REQUESTED** [review]

File: src/main/resources/icons/signIn/sign-in-logo-github.svg
Comment (Sa1to) [reply src/main/resources/icons/signIn/sign-in-logo-github.svg:1]: I don't see a problem. Buttons background color stays the same.

File: src/main/kotlin/com/sourcegraph/cody/chat/UIComponents.kt (hunk 3 above)
Comment (Sa1to) [reply src/main/kotlin/com/sourcegraph/cody/chat/UIComponents.kt:23]: Im not sure about including an image into an enum, seems more button related to me 😅

File: src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt (hunk 2 above)
Comment (Sa1to) [reply src/main/kotlin/com/sourcegraph/cody/chat/SignInWithSourcegraphPanel.kt]: It is old implementation. TBH i dont know if this is of any use, im just gonna remove it.

**Review by mkondratek — APPROVED** [review]
lgtm. pls verify for the last time that all SSO options work

**Pull request merged**